modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala

```diff
@@ -782,6 +781,234 @@ private[elasticsearch] final case class Ids[S](values: Chunk[String]) extends Id
+  intervals: Chunk[IntervalQuery],
+  filter: Option[IntervalFilter[S]]
+) extends IntervalQuery {
+  private[elasticsearch] def toJson: Json =
+    Obj(
+      "any_of" -> Obj(
+        "intervals" -> Arr(intervals.map(_.toJson): _*)
+      )
+    )
+}
+
+private[elasticsearch] final case class IntervalFuzzy(
+  term: String,
+  prefixLength: Option[Int],
+  transpositions: Option[Boolean],
+  fuzziness: Option[String],
+  analyzer: Option[String],
+  useField: Option[String]
+) extends IntervalQuery {
+  private[elasticsearch] def toJson: Json =
+    Obj(
+      "fuzzy" -> Obj(
+        Chunk(
+          Some("term" -> term.toJson),
+          prefixLength.map("prefix_length" -> _.toJson),
+          transpositions.map("transpositions" -> _.toJson),
+          fuzziness.map("fuzziness" -> _.toJson),
+          analyzer.map("analyzer" -> _.toJson),
+          useField.map("use_field" -> _.toJson)
+        ).flatten: _*
+      )
+    )
+}
+
+private[elasticsearch] final case class IntervalRange(
+  gt: Option[String],
+  gte: Option[String],
+  lt: Option[String],
+  lte: Option[String],
+  analyzer: Option[String],
+  useField: Option[String]
+) extends IntervalQuery {
+  private[elasticsearch] def toJson: Json =
+    Obj(
+      "range" -> Obj(
+        Chunk(
+          gt.map("gt" -> _.toJson),
+          gte.map("gte" -> _.toJson),
+          lt.map("lt" -> _.toJson),
+          lte.map("lte" -> _.toJson),
+          analyzer.map("analyzer" -> _.toJson),
+          useField.map("use_field" -> _.toJson)
+        ).flatten: _*
+      )
+    )
+}
+
+private[elasticsearch] final case class IntervalFilter[S](
+  after: Option[IntervalQuery] = None,
+  before: Option[IntervalQuery] = None,
+  containedBy: Option[IntervalQuery] = None,
+  containing: Option[IntervalQuery] = None,
+  notContainedBy: Option[IntervalQuery] = None,
+  notContaining: Option[IntervalQuery] = None,
+  notOverlapping: Option[IntervalQuery] = None,
+  overlapping: Option[IntervalQuery] = None,
+  script: Option[Json] = None
+) {
+  private[elasticsearch] def toJson: Json =
+    Obj(
+      Chunk(
+        after.map("after" -> _.toJson),
+        before.map("before" -> _.toJson),
+        containedBy.map("contained_by" -> _.toJson),
+        containing.map("containing" -> _.toJson),
+        notContainedBy.map("not_contained_by" -> _.toJson),
+        notContaining.map("not_containing" -> _.toJson),
+        notOverlapping.map("not_overlapping" -> _.toJson),
+        overlapping.map("overlapping" -> _.toJson),
+        script.map("script" -> _)
+      ).flatten: _*
+    )
+}
+
+sealed trait IntervalsQuery[S] extends ElasticQuery[S]
+
+private[elasticsearch] final case class Intervals[S](
+  field: String,
+  query: IntervalQuery
+) extends IntervalsQuery[S] {
+  self =>
+
+  private[elasticsearch] def toJson(fieldPath: Option[String]): Json =
+    Obj(
+      "intervals" -> Obj(
+        fieldPath.getOrElse(field) -> query.toJson
+      )
+    )
+}
```

Comment:
Only this should stay in this file.

Reply:
And all regarding IntervalQuery should be moved out (and renamed, as I previously commented).